subsys/bluetooth/host/l2cap.c

```diff
@@ -2027,37 +2037,66 @@ static int l2cap_chan_le_send(struct bt_l2cap_le_chan *ch,
 	return len;
 }
 
+/* return next netbuf fragment if present, also copy metadata */
+static struct net_buf *prepare_next_frag(struct net_buf *parent,
+					 struct l2cap_tx_meta_data *meta)
+{
+	/* this does an unref on `parent` */
+	struct net_buf *next = net_buf_frag_del(NULL, parent);
+
+	if (next) {
+		LOG_DBG("process next frag: %p -> %p", parent, next);
+
+		/* Copy the l2cap metadata ref to the next buffer in the chain.
+		 * We need it to figure out which channel the PDU has been sent
+		 * on, in order to either send the rest of the SDU or call the
+		 * application callback.
+		 */
+		l2cap_tx_meta_data(next) = meta;
```

Comment:
The naming scheme in this file is so bad. `meta` here is a pointer to a `meta_data` that is pointed to from `data` in the `meta` type. And the `meta_data` macro does not expand to the `meta_data` type but rather to the `meta_data data` pointer variable in `meta`.

:face_with_spiral_eyes: